pkg/cmd/cli/cmd/startbuild.go

```diff
@@ -479,31 +479,53 @@ func streamPathToBuild(git git.Repository, in io.Reader, out io.Writer, client o
 				if gitErr != nil {
 					return nil, fmt.Errorf("the directory %q is not a valid Git repository: %v", clean, gitErr)
 				}
-				pr, pw := io.Pipe()
-				go func() {
-					if err := git.Archive(clean, options.Commit, "tar.gz", pw); err != nil {
-						pw.CloseWithError(fmt.Errorf("unable to create Git archive of %q for build: %v", clean, err))
-					} else {
-						pw.CloseWithError(io.EOF)
-					}
-				}()
-				r = pr
+
+				// Create a temp directory to move the repo contents to
+				tempDirectory, err := ioutil.TempDir(os.TempDir(), "oc_cloning_"+options.Commit)
+				if err != nil {
+					return nil, err
+				}
+				defer os.RemoveAll(tempDirectory)
+
+				// We only want to grab the contents of the specified commit, with
+				// submodules included
+				cloneOptions := git.CloneOptions{
+					Recursive: true,
+					Shallow:   false,
+					// TODO: Should we output this at a loglevel?  If so, what level?
+					Quiet: true,
+				}
+
+				// Clone the repository to a temp directory for future tar-ing
+				if err := repo.CloneWithOptions(tempDirectory, clean, cloneOptions); err != nil {
+					return nil, err
+				}
+				if err := repo.Checkout(tempDirectory, options.Commit); err != nil {
```

Comment:
Will change it to 'commit', I meant that one but must have gotten them mixed up when coding.

commit is by default set to HEAD, so we can't have an empty string.  If we have an incorrect commit, line 503 should return an error